nautobot_ssot/jobs/base.py

```diff
@@ -47,17 +48,56 @@ class DataSyncBaseJob(BaseJob):
 
     dry_run = BooleanVar()
 
+    def load_source_adapter(self):
+        """TODO:."""
+        pass
+
+    def load_target_adapter(self):
+        """TODO:."""
+        pass
+
+    def calculate_diff(self):
```

Comment:
Similarly, this should either return the calculated diff, or consistently set it as `self.diff`, so that we can take advantage of https://github.com/networktocode/diffsync/pull/80 and avoid re-calculating the diff unnecessarily.